turbinia/workers/volatility.py

```diff
@@ -59,23 +60,40 @@ def run(self, evidence, result):
             evidence.local_path, evidence.profile, self.module,
             output_file_path).split()
 
-    result.log('Running vol as [{0:s}]'.format(' '.join(cmd)))
-    res = self.execute(
-        cmd, result, log_files=[debug_log], new_evidence=[output_evidence],
-        close=True)
+    result.log('Running volatility as [{0:s}]'.format(' '.join(cmd)))
+    res = self.execute(cmd, result, new_evidence=[output_evidence], close=True)
 
     if res == 0:
-      # Write the report to the output file.
-      with open(output_file_path, 'w') as fh:
-        fh.write(output_evidence.text_data.encode('utf-8'))
-
-      # Add the resulting evidence to the result object.
-      result.add_evidence(output_evidence, evidence.config)
-      result.report_data = output_evidence
-      summary = 'Volatility module {0} successfully ran.'.format(self.module)
+      # Get report data from the output file.
+      try:
+        file_size = os.stat(output_file_path).st_size
+      except OSError as exception:
+        msg = 'Cannot read output file {0:s}: {1!s}'.format(
+            output_file_path, exception)
+        summary = 'Volatility ran successfully, but no output file was created'
+        result.log(msg)
+        result.close(self, success=False, status=summary)
+        return result
+
+      if file_size > MAX_REPORT_SIZE:
+        result.log(
+            'Volatility report output size ({0:d}) is greater than max report '
+            'size ({1:d}). Truncating report to max size'.format(
+                file_size, MAX_REPORT_SIZE))
+        summary = (
+            'Volatility module {0:s} successfuly ran (report truncated)'.format(
+                self.module))
+      else:
+        summary = 'Volatility module {0:s} successfully ran'.format(self.module)
+
+      with open(output_file_path, 'rb') as fh:
```

Comment:
Why not the following approach without stat

first read fh.read(MAX_REPORT_SIZE)
then read 1 byte if len(report_data) == MAX_REPORT_SIZE, if that byte contains data provide the truncation warning

Reply:
I'm not sure if I see much of an advantage doing it that way.  Unless there's a compelling reason, I think I'll keep it the way it is for now.

Reply:
this was more relevant, when you cannot use stat